static/js/components/AudioPlayer_test.js

```diff
@@ -0,0 +1,65 @@
+import React from "react"
+import { Provider } from "react-redux"
+import { assert } from "chai"
+import { mount } from "enzyme"
+
+import AudioPlayer from "./AudioPlayer"
+import Amplitude from "amplitudejs"
+import {
+  SET_CURRENTLY_PLAYING_AUDIO,
+  setCurrentlyPlayingAudio
+} from "../actions/audio"
+import IntegrationTestHelper from "../util/integration_test_helper"
+
+describe("AudioPlayer", () => {
+  let helper, wrapper, dispatchThen
+
+  const renderAudioPlayer = (props = {}, store) =>
+    mount(
+      <Provider store={store}>
+        <AudioPlayer {...props} />
+      </Provider>
+    )
+  const exampleAudio = {
+    title:       "Test Title",
+    description: "Test Description",
+    url:
+      "https://chtbl.com/track/F9DD6B/cdn.simplecast.com/audio/2c64ac/2c64ace6-baf4-4e86-b527-445e611e6a31/aa0ca88f-3c4c-4d33-9897-36e45c43e012/film-is-for-everyone-with-prof-david-thorburn_tc.mp3"
+  }
+
+  beforeEach(() => {
+    helper = new IntegrationTestHelper()
+    wrapper = renderAudioPlayer({}, helper.store)
+    dispatchThen = helper.store.createDispatchThen(state => state.audio)
+    dispatchThen(setCurrentlyPlayingAudio(exampleAudio), [
+      SET_CURRENTLY_PLAYING_AUDIO
+    ])
```

Comment:
I think you may be able to simplify some of this code if you use the `reduxQueryRenderer` in `IntegrationTestHelper`, you won't have to set up the `dispatchThen` thing for instance

Reply:
So I did this in the PodcastEpisodeCard tests and tried the same here, but I couldn't get the last test to work right with reduxQueryRenderer for some reason.  Amplitude is never initialized when the test is run in this way, but it is if I create the dispatch manually.

Reply:
how were you setting up the redux state when you were using it? after you create the renderer there is a second argument you can pass to `render` called `beforeRenderActions`, you can pass a list of actions that will be dispatched to the store before the component is mounted. you should be able to use it like

```js
const { wrapper } = await render({}, [setCurrentlyPlayingAudio(exampleAudio)])
```

I think you can look at `components/AddToListDialog_test.js` for an example of this

Reply:
Ah yea that worked.  Thanks!